### ISSUE TITLE
cluster.stop() and cluster.restart(cluster_id) methods for AzureVMCLuster class

### ISSUE THREAD

**Issue by manuelreyesgomez** [dask/dask-cloudprovider#278]
The close() method deletes the cluster from Azure

But what if the user has done work on it and has saved work into the scheduler and wants to later come back and pick up where it stopped

It would be great if the user could stop the cluster which would stop all VMs but not delete them and that the user could later come back and restart the cluster by passing its ID

**Comment by jacobtomlinson** [comment]
Absolutely! This work is in the pipeline as part of the integration with [dask-ctl](https://github.com/dask-contrib/dask-ctl).

Today it is possible to avoid the cluster being shutdown by setting the `shutdown_on_close` attribute to `False` on the cluster manager. However this is not currently advertised in this project as there is no way to reconstruct the cluster object with an existing cluster. Do if you deleted the Python object without the cluster being deleted on Azure you would have to clean it up manually.
